### PR TITLE
Implement PAUSER priv levels

### PR DESCRIPTION
**Pull request by jhand2**
Add the concept of a PL0 PAUSER to the image format and expose this information to RT firmware.

Additionally, add scaffolding for a status mailbox comment to report back firmware information. For now it only reports the PL0 PAUSER.

### CODE REVIEW

File: runtime/src/status.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        resp
+    }
+}
+
+#[repr(C)]
+#[derive(AsBytes, FromBytes)]
+pub struct StatusResp {
+    pub chksum: i32,
+    pub pl0_pauser: u32,
+    // TODO: Decide what other information to report for general firmware
```
Comment (bluegate010) [inline runtime/src/status.rs:28]: Maybe "current PAUSER"?

Comment (jhand2) [reply runtime/src/status.rs:28]: Planning to just leave this for this PR. There is a tracking item in the smartsheet for a full status command. I just added this to have a quick way to test this change.

**Review by mhatrevi — DISMISSED** [review]

File: runtime/src/main.rs [hunk 2]
```diff
@@ -36,14 +36,15 @@ const BANNER: &str = r#"
...
     if let Some(mut fht) = caliptra_common::FirmwareHandoffTable::try_load() {
-        let mut drivers = unsafe { Drivers::new_from_registers(&mut fht) };
-
-        // Indicator to SOC that RT firmware is ready
-        drivers
-            .soc_ifc
-            .regs_mut()
-            .cptra_flow_status()
-            .write(|w| w.ready_for_runtime(true));
+        let mut drivers = match unsafe { Drivers::new_from_registers(&mut fht) } {
```
Comment (jlmahowa-amd) [inline runtime/src/main.rs:39]: Looks like a merge conflict. Did you intent to remove the ready_for_runtime on the left? I don't see it re-added elsewhere.

**Review by jlmahowa-amd — COMMENTED** [review]
Will there need to be a hw-model update to provide the right PAUSER eventually?

File: runtime/src/main.rs (hunk 2 above)
Comment (jhand2) [reply runtime/src/main.rs:39]: I was thinking this was a duplicate of https://github.com/chipsalliance/caliptra-sw/blob/012e20c20521ed3a56956a5b0249384365b23f67/runtime/src/lib.rs#L201C59-L201C59

I didn't realize boot_status and flow_status are different. I'll add this back.

**Comment by jhand2** [comment]
> Will there need to be a hw-model update to provide the right PAUSER eventually?

Ya. I'll do this in a follow-up PR

File: runtime/src/main.rs (hunk 2 above)
Comment (jhand2) [reply runtime/src/main.rs:39]: moved to `handle_mailbox_commands` alongside the boot status reporting.

**Review by jlmahowa-amd — DISMISSED** [review]

**Review by korran — DISMISSED** [review]
Approved based on vmhatre's earlier approval and a harmless change:

```diff
diff --git a/runtime/src/lib.rs b/runtime/src/lib.rs
index 60ea5eb2..a7db5cb3 100644
--- a/runtime/src/lib.rs
+++ b/runtime/src/lib.rs
@@ -218,6 +218,12 @@ fn handle_command(drivers: &mut Drivers) -> CaliptraResult<MboxStatusE> {
 }
 
 pub fn handle_mailbox_commands(drivers: &mut Drivers) {
+    // Indicator to SOC that RT firmware is ready
+    drivers
+        .soc_ifc
+        .regs_mut()
+        .cptra_flow_status()
+        .write(|w| w.ready_for_runtime(true));
     caliptra_drivers::report_boot_status(RtBootStatus::RtReadyForCommands.into());
     loop {
         wait_for_cmd(&mut drivers.mbox);
```

**Review by rusty1968 — DISMISSED** [review]

**Review by rusty1968 — DISMISSED** [review]

**Review by rusty1968 — APPROVED** [review]

**Review by korran — APPROVED** [review]

**Pull request merged**